File: 119-pascal's-triangle-ll.java
class Solution {
    public List<Integer> getRow(int rowIndex) {
        List<Integer> triangle = new ArrayList<>();
        triangle.add(1);

        for(int i=0;i<rowIndex;i++){
            List<Integer> prev =new ArrayList<>();
            for(int j=0;j<triangle.size()-1;j++){
                prev.add(triangle.get(j) + triangle.get(j+1));
            }
            triangle.clear();
            triangle.addAll(prev);
            triangle.add(0,1);
            triangle.add(1);
        }
        return triangle;
    }
}

File: 119-pascal's-triangle-ll.py
class Solution:
    def getRow(self, rowIndex: int) -> List[int]:
        triangle = [[1]]
        for i in range(1,rowIndex+1):
            row = []
            for j in range(i+1):
                if j==0 or j==i:
                    row.append(1)
                else:
                    val = triangle[0][j-1] + triangle[0][j]
                    row.append(val)
            triangle[0] = row
        return triangle[0]